packages/@aws-cdk/aws-servicecatalog/lib/product-stack.ts

```diff
@@ -50,6 +52,33 @@ export class ProductStack extends cdk.Stack {
     return cdk.Lazy.uncachedString({ produce: () => this._templateUrl });
   }
 
+  /**
+   * Fetch the asset bucket.
+   *
+   * @internal
+   */
+  public _getAssetBucket(): IBucket | undefined {
+    return this.assetBucket;
+  }
```

Comment:
Why not make the property public readonly instead?

Reply:
If we make it public readonly we get the following error hint in the `_setAssetbucket` method: 
`Cannot assign to 'assetBucket' because it is a read-only property`

Making this property public readonly would force us to initialize the property in the constructor, which we are unable to do directly. We are unable to do this because, by design, we want to conditionally define the `assetBucket` property only in the case where the user has a product stack that uses assets. If the user has a product stack without assets, we would like for the `assetBucket` property to stay as undefined.

Reply:
Why have this property here at all? It only seems to exist here to copy the value from the synthesizer up to the portfolio.

Can we skip the middle man in some way?

Reply:
Correct, the ultimate goal here is to get the asset bucket to the portfolio, so that the portfolio is able to share its buckets with any account the portfolio is shared with. For context on what we can do, take a look at: https://github.com/aws/aws-cdk/pull/21508#discussion_r942536461

Under the scenario where the user creates the product stack asset bucket in the application stack, we can skip the middle man by having the user pass in the product stack as a prop to each product stack AND the portfolio. However, would this provide users with an ideal experience? In the case where the user passes in the asset bucket to the portfolio and not the product stacks (or vice versa), how could we detect this?